test/functional/p2p_ibd_txrelay.py

```diff
@@ -22,7 +50,50 @@ def set_test_params(self):
             ["-minrelaytxfee={}".format(NORMAL_FEE_FILTER)],
         ]
 
-    def run_test(self):
+    def test_inv_ibd(self):
+        self.log.info("Check that nodes don't request txn while still in IBD")
+        peer = self.nodes[0].add_p2p_connection(P2PInterface())
+        peer.send_message(msg_inv([CInv(t=MSG_WTX, h=0xffaa)]))
+        # As tx-announcing connection is inbound, wait for the inbound delay applied by requester.
+        sleep(INBOUND_PEER_TX_DELAY)
+        peer.sync_with_ping()
+        with p2p_lock:
+            assert_equal(peer.tx_getdata_count, 0)
+        peer.peer_disconnect()
+        peer.wait_for_disconnect()
+
+    def test_orphan_ibd(self):
+        self.log.info("Check that nodes don't request orphan parent while still in IBD")
+
+        # Build an orphan parent tx
+        witness_program = CScript([OP_TRUE])
+        witness_hash = sha256(witness_program)
+        script_pubkey = CScript([OP_0, witness_hash])
+
+        tx_parent = CTransaction()
+        tx_parent.vin.append(CTxIn(outpoint=COutPoint(uint256_from_str(b"23c70ed7c0506e9178fc1a987f40a33946d4ad4c962b5ae3a52546da53af0c5c"),
+ 0)))
+        tx_parent.vout.append(CTxOut(nValue=50 * COIN - 12000, scriptPubKey=script_pubkey))
+        tx_parent.calc_sha256()
+
+        # Build an orphan tx
+        tx_orphan = CTransaction()
+        tx_orphan.vin.append(CTxIn(outpoint=COutPoint(tx_parent.sha256, 0)))
+        tx_orphan.vout = [CTxOut(nValue=10 * COIN, scriptPubKey=script_pubkey)]
+        tx_orphan.calc_sha256()
+
+        peer = self.nodes[0].add_p2p_connection(P2PInterface())
+        peer.send_message(msg_tx(tx_orphan))
+        # As tx-announcing connection is inbound, wait for the inbound delay applied by requester.
+        # Also apply the txid delay as orphan parent are requested in the legacy way.
+        sleep(INBOUND_PEER_TX_DELAY+TXID_RELAY_DELAY)
```

Comment:
8afa4fa s/parent are/parents are/
```suggestion
        sleep(INBOUND_PEER_TX_DELAY + TXID_RELAY_DELAY)
```